File: test/fixtures/small-java/Greeter.java
/** Greets people with a configurable prefix. */
public class Greeter {
  private final String prefix;

  public Greeter(String prefix) {
    this.prefix = prefix;
  }

  /** Returns the greeting for one person. */
  public String greet(String name) {
    return format(name);
  }

  private String format(String name) {
    return prefix + ", " + name + "!";
  }

  public static class Builder {
    private String prefix = "Hello";

    public Builder prefix(String value) {
      this.prefix = value;
      return this;
    }

    public Greeter build() {
      return new Greeter(prefix);
    }
  }
}

File: test/fixtures/small-go/greeter.go
package main

// Greeter greets people with a configurable prefix.
type Greeter struct {
	prefix string
}

// NewGreeter builds a Greeter with the given prefix.
func NewGreeter(prefix string) *Greeter {
	return &Greeter{prefix: prefix}
}

// Greet returns the greeting for one person.
func (g *Greeter) Greet(name string) string {
	return g.format(name)
}

func (g *Greeter) format(name string) string {
	return g.prefix + ", " + name + "!"
}
